src/UnderlineNav2/UnderlineNav.test.tsx

```diff
@@ -112,6 +114,14 @@ describe('UnderlineNav', () => {
     const user = userEvent.setup()
     await user.click(item)
     expect(onSelect).toHaveBeenCalledTimes(1)
+
+    if (REACT_VERSION_LATEST) {
+      expect(spy).not.toHaveBeenCalled()
+    } else {
+      // Warning: It looks like you're using the wrong act() around your test interactions
+      expect(spy).toHaveBeenCalledTimes(2)
+    }
+    spy.mockRestore()
```

Comment:
Could you tell me a little bit more about this test? Not sure if I understand it well. Thank you 🙏🏻

Reply:
@broccolinisoup would you have time to pair up on this next week? Would love to walk through this case with you, it's super strange 🤔

I've encountered this with a couple components and it seems to happen where there is a mix of snapshot testing from the test helpers and react testing library. I don't think anything is wrong with the test itself and instead it's the error showing up due to this mix-and-match of `react-test-renderer` and `@testing-library/react`. 

In `@testing-library/react`, there is an "act-compat" module that seems to be picking the wrong thing and causing this error to get thrown but I've been unable to reproduce what specifically is causing it.